TcpIpInput.cpp

```diff
@@ -2,15 +2,59 @@
 
 
 
-TcpIpInput::TcpIpInput(short port){
-    mPort = port;
-    std::string portAsString = std::to_string(mPort);
-    std::string fullCommand = "nc -l "+ portAsString;
-    const char* convertedCommand = fullCommand.c_str();
-    static_cast<void>(std::system(convertedCommand)); //execute command
+TcpIpInput::TcpIpInput(short port, unsigned short domain, unsigned int type){
 
+    mAddress.sin_port = htons(port);
+    mAddress.sin_family = domain;
+    mAddress.sin_addr.s_addr = type;
+    int opt = 1;
+
+    // Creating socket file descriptor
+    if ((mServerFile = socket(domain, SOCK_STREAM, 0)) == 0)
+    {
+        perror("socket failed");
+        exit(EXIT_FAILURE);
+    }
+
+    // Forcefully attaching socket to the port
+    if (setsockopt(mServerFile, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT,
+                                                  &opt, sizeof(opt)))
+    {
+        perror("setsockopt");
+        exit(EXIT_FAILURE);
+    }
+
+
+    // Forcefully attaching socket to the port
+    if (bind(mServerFile,(struct sockaddr *)&mAddress,sizeof(mAddress))<0)
+    {
+        perror("bind failed");
+        exit(EXIT_FAILURE);
+    }
 }
 
 std::string TcpIpInput::getData(){
-    return "defualtValue";
+    int newSocket, valRead;
+    int addrLen = sizeof(this->mAddress);
+    if (listen(this->mServerFile, 3) < 0)
+    {
+        perror("listen");
```

Comment:
What is perror?

Reply:
it;s from stdio.h. It prints understandable code error. I'll add comment

Reply:
no need, it's fine. I just have never seen it before. Though, if you this is what you want, then perhaps you should use it in the whole project for printing errors, for consistency.

Reply:
OK